src/BatteryStateOfChargeService.h

```diff
@@ -19,4 +22,12 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Recommend adding necessary variables here
+
+    QTime previousTime_;
+    QTime newTime_;
+    double previousCurrent_;
+    double deltaTime_;
+    double aveCurrent_;
```

Comment:
Normally, the abbreviation for average is `avg` not `ave`, I think `ave` to more people would imply avenue haha, so I think `avg` would be better.